WIP - Prepare for Quarkus 1.2

gsmet opened the pull request:
This PR should not be merged. I created it to be able to discuss with the GraalVM team.

lburgazzoli commented:
GraalVM 19.3.0.2 ?

gsmet commented:
Yes, this is with GraalVM 19.3.0.2 but the `beans` ITs are still failing.

I wanted to have a reproducer for the GraalVM team.

gsmet commented:
@lburgazzoli could you test this PR with Quarkus master and the just released GraalVM 19.3.1 ?

lburgazzoli commented:
I'm testing this PR and so far I've found some issues on tests that were previously working:

1. On some tests, `BaseMainSupport.loadConfigurations` triggers the following error and I think the stack trace is unrelated as the same copde works on other tests but I'm unable to identify the error as the stracktrace is misleading:

```
[camel-quarkus-integration-test-http-1.1.0-SNAPSHOT-runner:58666]     analysis:  30,537.78 ms
Fatal error: com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing com.oracle.svm.reflect.ClassLoader_defineClass_c6c343b4d6dc22ca64eb2d8503b13ac9c340dcb3.invoke(java.lang.Object, java.lang.Object[])
Parsing context:
    parsing java.lang.reflect.Method.invoke(Method.java:498)
    parsing org.apache.camel.support.ObjectHelper.invokeMethod(ObjectHelper.java:182)
    parsing org.apache.camel.main.BaseMainSupport.loadConfigurations(BaseMainSupport.java:467)
    parsing org.apache.camel.main.BaseMainSupport.postProcessCamelContext(BaseMainSupport.java:522)
```

2. the `jdbc` tests fails with:

```
[camel-quarkus-integration-test-jdbc-1.1.0-SNAPSHOT-runner:64494]     analysis:  26,304.26 ms
2 fatal errors detected:
Fatal error: com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing com.oracle.svm.reflect.Engine_getInstance_e48f9e2f1a3bb2d3099996112ca26ffbbf8b77da.invoke(java.lang.Object, java.lang.Object[])
Parsing context:
    parsing java.lang.reflect.Method.invoke(Method.java:498)
    parsing javax.enterprise.util.AnnotationLiteral.invoke(AnnotationLiteral.java:288)
    parsing javax.enterprise.util.AnnotationLiteral.getMemberValue(AnnotationLiteral.java:276)
    parsing javax.enterprise.util.AnnotationLiteral.hashCode(AnnotationLiteral.java:246)
    parsing java.util.concurrent.CopyOnWriteArrayList.hashCode(CopyOnWriteArrayList.java:1062)

    ---
Caused by: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method org.h2.engine.Engine.getInstance() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
```

3. Lot of test are start failing in native mode because the `native-image` complains about the lack of `com/fasterxml/jackson/databind/JsonSerializer` so I added `quarkus-jackson` as workaround but that smells to me.

gsmet commented:
If you are using Jackson, you need the Jackson extension. 

As for the other issues, they need to be sorted out. But I don’t understand how we got there. 

@geoand @mkouba could you try to have a look?

> Le 15 janv. 2020 à 13:50, Luca Burgazzoli <notifications@github.com> a écrit :
> 
> I'm testing this PR and so far I've found some issues on tests that were previously working:
> 
> On some tests, BaseMainSupport.loadConfigurations triggers the following error and I think the stack trace is unrelated as the same copde works on other tests but I'm unable to identify the error as the stracktrace is misleading:
> [camel-quarkus-integration-test-http-1.1.0-SNAPSHOT-runner:58666]     analysis:  30,537.78 ms
> Fatal error: com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing com.oracle.svm.reflect.ClassLoader_defineClass_c6c343b4d6dc22ca64eb2d8503b13ac9c340dcb3.invoke(java.lang.Object, java.lang.Object[])
> Parsing context:
>     parsing java.lang.reflect.Method.invoke(Method.java:498)
>     parsing org.apache.camel.support.ObjectHelper.invokeMethod(ObjectHelper.java:182)
>     parsing org.apache.camel.main.BaseMainSupport.loadConfigurations(BaseMainSupport.java:467)
>     parsing org.apache.camel.main.BaseMainSupport.postProcessCamelContext(BaseMainSupport.java:522)
> the jdbc tests fails with:
> [camel-quarkus-integration-test-jdbc-1.1.0-SNAPSHOT-runner:64494]     analysis:  26,304.26 ms
> 2 fatal errors detected:
> Fatal error: com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing com.oracle.svm.reflect.Engine_getInstance_e48f9e2f1a3bb2d3099996112ca26ffbbf8b77da.invoke(java.lang.Object, java.lang.Object[])
> Parsing context:
>     parsing java.lang.reflect.Method.invoke(Method.java:498)
>     parsing javax.enterprise.util.AnnotationLiteral.invoke(AnnotationLiteral.java:288)
>     parsing javax.enterprise.util.AnnotationLiteral.getMemberValue(AnnotationLiteral.java:276)
>     parsing javax.enterprise.util.AnnotationLiteral.hashCode(AnnotationLiteral.java:246)
>     parsing java.util.concurrent.CopyOnWriteArrayList.hashCode(CopyOnWriteArrayList.java:1062)
> 
>     ---
> Caused by: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method org.h2.engine.Engine.getInstance() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
> To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
> Lot of test are start failing in native mode because the native-image complains about the lack of com/fasterxml/jackson/databind/JsonSerializer so I added quarkus-jackson as workaround but that smells to me.
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

geoand commented:
Sure, I'll check it in a while

gsmet commented:
Oh and about H2, are you using it in a native image? If you are using the JDBC driver, you should have the H2 extension. 

Are they new tests from new extensions?

It’s really weird we have these issues now. GraalVM 19.3.1 is not that different from the previous version. 

> Le 15 janv. 2020 à 13:50, Luca Burgazzoli <notifications@github.com> a écrit :
> 
> I'm testing this PR and so far I've found some issues on tests that were previously working:
> 
> On some tests, BaseMainSupport.loadConfigurations triggers the following error and I think the stack trace is unrelated as the same copde works on other tests but I'm unable to identify the error as the stracktrace is misleading:
> [camel-quarkus-integration-test-http-1.1.0-SNAPSHOT-runner:58666]     analysis:  30,537.78 ms
> Fatal error: com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing com.oracle.svm.reflect.ClassLoader_defineClass_c6c343b4d6dc22ca64eb2d8503b13ac9c340dcb3.invoke(java.lang.Object, java.lang.Object[])
> Parsing context:
>     parsing java.lang.reflect.Method.invoke(Method.java:498)
>     parsing org.apache.camel.support.ObjectHelper.invokeMethod(ObjectHelper.java:182)
>     parsing org.apache.camel.main.BaseMainSupport.loadConfigurations(BaseMainSupport.java:467)
>     parsing org.apache.camel.main.BaseMainSupport.postProcessCamelContext(BaseMainSupport.java:522)
> the jdbc tests fails with:
> [camel-quarkus-integration-test-jdbc-1.1.0-SNAPSHOT-runner:64494]     analysis:  26,304.26 ms
> 2 fatal errors detected:
> Fatal error: com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing com.oracle.svm.reflect.Engine_getInstance_e48f9e2f1a3bb2d3099996112ca26ffbbf8b77da.invoke(java.lang.Object, java.lang.Object[])
> Parsing context:
>     parsing java.lang.reflect.Method.invoke(Method.java:498)
>     parsing javax.enterprise.util.AnnotationLiteral.invoke(AnnotationLiteral.java:288)
>     parsing javax.enterprise.util.AnnotationLiteral.getMemberValue(AnnotationLiteral.java:276)
>     parsing javax.enterprise.util.AnnotationLiteral.hashCode(AnnotationLiteral.java:246)
>     parsing java.util.concurrent.CopyOnWriteArrayList.hashCode(CopyOnWriteArrayList.java:1062)
> 
>     ---
> Caused by: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method org.h2.engine.Engine.getInstance() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
> To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
> Lot of test are start failing in native mode because the native-image complains about the lack of com/fasterxml/jackson/databind/JsonSerializer so I added quarkus-jackson as workaround but that smells to me.
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

lburgazzoli commented:
About jackson the tess were working before so either something has changed on quarkus side I guess

lburgazzoli commented:
> Oh and about H2, are you using it in a native image? If you are using the JDBC driver, you should have the H2 extension.

we have quarkus-test-h2

lburgazzoli commented:
> Are they new tests from new extensions?

No they are old extensions

geoand commented:
> About jackson the tests were working before so either something has changed on quarkus side I guess

Which version of Quarkus were they working for?

lburgazzoli commented:
> > About jackson the tests were working before so either something has changed on quarkus side I guess
> 
> Which version of Quarkus were they working for?

1.1.0.Final

geoand commented:
I'll start looking into it

mkouba commented:
@lburgazzoli I'm not an expert here but if you look at `io.quarkus.jdbc.h2.runtime.graal.Engine.createSession(ConnectionInfo)` there is a note: "H2 database compiled into a native-image is only functional as a client: can't create an Embedded Database Session". So does it make sense to run the `CamelJdbcIT`?

lburgazzoli commented:
We do use it as client, we use quarkus-test-h2 as test dependency to provide the server part

lburgazzoli commented:
Interesting enough, the same test show different stack trace when the native image fails depending on the jvm, so i.e. the sjms test:

- JDK 1.8
```
Fatal error: com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing com.oracle.svm.reflect.ClassLoader_defineClass_c6c343b4d6dc22ca64eb2d8503b13ac9c340dcb3.invoke(java.lang.Object, java.lang.Object[]) 
Parsing context:
	parsing java.lang.reflect.Method.invoke(Method.java:498)
	parsing org.apache.camel.support.ObjectHelper.invokeMethod(ObjectHelper.java:184)
	parsing org.apache.camel.main.BaseMainSupport.loadConfigurations(BaseMainSupport.java:467)
	parsing org.apache.camel.main.BaseMainSupport.postProcessCamelContext(BaseMainSupport.java:522)
	parsing org.apache.camel.quarkus.core.CamelMain.postProcessCamelContext(CamelMain.java:59)
	parsing org.apache.camel.quarkus.core.CamelMain.doStart(CamelMain.java:49)
	parsing org.apache.camel.support.service.ServiceSupport.start(ServiceSupport.java:117)
	parsing org.apache.camel.quarkus.core.CamelMainRecorder.start(CamelMainRecorder.java:92)
	parsing io.quarkus.deployment.steps.Main$start80.deploy_0(Main$start80.zig:97)
	parsing io.quarkus.deployment.steps.Main$start80.deploy(Main$start80.zig:36)
	parsing io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:188)
	parsing io.quarkus.runtime.Application.start(Application.java:87)
	parsing io.quarkus.runtime.Application.run(Application.java:210)
	parsing io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:41)
	parsing com.oracle.svm.core.JavaMainWrapper.runCore(JavaMainWrapper.java:151)
	parsing com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:186)
	parsing com.oracle.svm.core.code.IsolateEnterStub.JavaMainWrapper_run_5087f5482cc9a6abc971913ece43acb471d2631b(generated:0)

	at com.oracle.graal.pointsto.util.AnalysisError.parsingError(AnalysisError.java:138)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.doParse(MethodTypeFlow.java:327)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureParsed(MethodTypeFlow.java:300)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.addContext(MethodTypeFlow.java:107)
	at com.oracle.graal.pointsto.DefaultAnalysisPolicy$DefaultVirtualInvokeTypeFlow.onObservedUpdate(DefaultAnalysisPolicy.java:191)
	at com.oracle.graal.pointsto.flow.TypeFlow.notifyObservers(TypeFlow.java:343)
	at com.oracle.graal.pointsto.flow.TypeFlow.update(TypeFlow.java:385)
	at com.oracle.graal.pointsto.BigBang$2.run(BigBang.java:511)
	at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$execute$0(CompletionExecutor.java:171)
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.ClassLoader.defineClass(String, byte[], int, int) is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
```

- JDK 11
```
Fatal error: com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing com.oracle.svm.reflect.ClassLoader_defineClass_c6c343b4d6dc22ca64eb2d8503b13ac9c340dcb3.invoke(java.lang.Object, java.lang.Object[]) 
Parsing context:
	parsing java.lang.reflect.Method.invoke(Method.java:566)
	parsing javax.enterprise.util.AnnotationLiteral.invoke(AnnotationLiteral.java:288)
	parsing javax.enterprise.util.AnnotationLiteral.getMemberValue(AnnotationLiteral.java:276)
	parsing javax.enterprise.util.AnnotationLiteral.hashCode(AnnotationLiteral.java:246)
	parsing java.util.HashMap.hash(HashMap.java:339)
	parsing java.util.HashMap.get(HashMap.java:552)
	parsing com.oracle.svm.core.jdk.Target_sun_util_locale_provider_LocaleProviderAdapter.forType(LocaleSubstitutions.java:104)
	parsing sun.util.locale.provider.LocaleProviderAdapter.forJRE(LocaleProviderAdapter.java:202)
	parsing java.text.DecimalFormatSymbols.getInstance(DecimalFormatSymbols.java:182)
	parsing java.util.Formatter.getZero(Formatter.java:2437)
	parsing java.util.Formatter.<init>(Formatter.java:1956)
	parsing java.util.Formatter.<init>(Formatter.java:1978)
	parsing java.lang.String.format(String.java:2897)
	parsing java.lang.Character.toChars(Character.java:8573)
	parsing java.lang.StringLatin1.toLowerCaseEx(StringLatin1.java:434)
	parsing java.lang.StringLatin1.toLowerCase(StringLatin1.java:395)
	parsing java.lang.String.toLowerCase(String.java:2503)
	parsing java.lang.String.toLowerCase(String.java:2527)
	parsing com.oracle.svm.core.option.SubstrateOptionsParser.parseLong(SubstrateOptionsParser.java:436)
	parsing com.oracle.svm.core.option.XOptions.parseFromValueString(XOptions.java:190)
	parsing com.oracle.svm.core.option.XOptions.parseWithNameAndPrefix(XOptions.java:182)
	parsing com.oracle.svm.core.option.XOptions.parse(XOptions.java:156)
	parsing com.oracle.svm.core.option.RuntimeOptionParser.parseAndConsumeAllOptions(RuntimeOptionParser.java:78)
	parsing com.oracle.svm.jni.functions.JNIInvocationInterface$Exports.JNI_CreateJavaVM(JNIInvocationInterface.java:151)
	parsing com.oracle.svm.core.code.IsolateEnterStub.JNIInvocationInterface$Exports_JNI_CreateJavaVM_8df18e9ea6eb1ed7b1c899a0bbc578da0e2cc0ad(generated:0)

	at com.oracle.graal.pointsto.util.AnalysisError.parsingError(AnalysisError.java:138)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.doParse(MethodTypeFlow.java:327)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureParsed(MethodTypeFlow.java:300)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.addContext(MethodTypeFlow.java:107)
	at com.oracle.graal.pointsto.DefaultAnalysisPolicy$DefaultVirtualInvokeTypeFlow.onObservedUpdate(DefaultAnalysisPolicy.java:191)
	at com.oracle.graal.pointsto.flow.TypeFlow.notifyObservers(TypeFlow.java:343)
	at com.oracle.graal.pointsto.flow.TypeFlow.update(TypeFlow.java:385)
	at com.oracle.graal.pointsto.BigBang$2.run(BigBang.java:511)
	at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$execute$0(CompletionExecutor.java:171)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Caused by: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.ClassLoader.defineClass(String, byte[], int, int) is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
```

lburgazzoli commented:
so as discovered by @geoand , the jackson issue is caused by the flag `reportErrorsAtRuntime` being set to `true` which is a consequence of the other errors

gsmet commented:
I just rebased.

gsmet commented on integration-tests/aws/pom.xml:
```diff
@@ -152,7 +152,6 @@
                                     <enableHttpUrlHandler>true</enableHttpUrlHandler>
                                     <enableServer>false</enableServer>
                                     <dumpProxies>false</dumpProxies>
-                                    <enableJni>true</enableJni>
```
Hmmm, so, JNI is not enabled by default in 19.2 and we will support both. So if you need JNI, you should add a `JniBuildItem` in your extension.

lburgazzoli replied:
we'll have a look

gsmet commented on integration-tests/http/pom.xml:
```diff
@@ -122,13 +135,30 @@
+
+                                        TODO: investigate what is the real issue
+                                    -->
+                                    <reportErrorsAtRuntime>true</reportErrorsAtRuntime>
```
I don't think we should have that ones in this branch because it only hides the errors and we need to be able to diagnose them.

lburgazzoli replied:
we just need to revert my commit

geoand replied:
Yeah, I think we should do that. It will it easier to reproduce the actual problem

lburgazzoli replied:
do it :)

geoand commented:
I am pretty sure that commit ee62404b93569d07fa119a10d6a58ee42fbf0560 now needs to be removed (since we are supporting both GraalVM 19.2 and 19.3) in Quarkus. Right @gsmet ?

lburgazzoli commented:
@gsmet can you please rebase ?

geoand commented:
@gsmet do you want me to rebase? If so, you'll need to grant me write permission to your fork

gsmet commented:
I'm running the tests with 1.2.0.CR1 as we speak.

geoand commented:
If you have pushed the change somewhere, I would like to launch my custom Github actions job as well

gsmet commented:
I will push once the tests are OK on my machine :). I'll ping you.

geoand commented:
OK, thanks

gsmet commented:
I just rebased and updated to 1.2.0.CR1.

@lburgazzoli CI reports a weird error: `[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.2.0.CR1:build (default) on project camel-quarkus-integration-test-servlet: Failed to build a runnable JAR: Failed to augment application classes: Property user.country not found -> [Help 1]`. Do you have any idea of what's going on?

gsmet commented:
@geoand ping.

lburgazzoli commented:
The stack trace is:

```
2020-01-21T11:12:21.3603975Z [INFO] Running org.apache.camel.quarkus.component.xml.deployment.CamelDevModeTest
2020-01-21T11:12:21.8279540Z 11:12:21,813 ERROR [io.qua.dev.DevModeMain] Failed to start quarkus: java.util.NoSuchElementException: Property user.country not found
2020-01-21T11:12:21.8280322Z 	at io.smallrye.config.SmallRyeConfig.propertyNotFound(SmallRyeConfig.java:206)
2020-01-21T11:12:21.8280457Z 	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:96)
2020-01-21T11:12:21.8280577Z 	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:80)
2020-01-21T11:12:21.8280713Z 	at io.quarkus.runtime.configuration.ConfigExpander.accept(ConfigExpander.java:55)
2020-01-21T11:12:21.8280841Z 	at io.quarkus.runtime.configuration.ConfigExpander.accept(ConfigExpander.java:15)
2020-01-21T11:12:21.8280963Z 	at org.wildfly.common.expression.ExpressionNode.emit(ExpressionNode.java:42)
2020-01-21T11:12:21.8281084Z 	at org.wildfly.common.expression.CompositeNode.emit(CompositeNode.java:42)
2020-01-21T11:12:21.8281206Z 	at org.wildfly.common.expression.Expression.evaluateException(Expression.java:75)
2020-01-21T11:12:21.8281319Z 	at org.wildfly.common.expression.Expression.evaluate(Expression.java:89)
2020-01-21T11:12:21.8281445Z 	at io.quarkus.runtime.configuration.ExpandingConfigSource.expandValue(ExpandingConfigSource.java:101)
2020-01-21T11:12:21.8281573Z 	at io.quarkus.runtime.configuration.ExpandingConfigSource.expand(ExpandingConfigSource.java:69)
2020-01-21T11:12:21.8281697Z 	at io.quarkus.runtime.configuration.ExpandingConfigSource.getValue(ExpandingConfigSource.java:47)
2020-01-21T11:12:21.8282127Z 	at io.quarkus.runtime.configuration.DeploymentProfileConfigSource.getValue(DeploymentProfileConfigSource.java:80)
2020-01-21T11:12:21.8282273Z 	at io.smallrye.config.SmallRyeConfig.getRawValue(SmallRyeConfig.java:121)
2020-01-21T11:12:21.8282394Z 	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:84)
2020-01-21T11:12:21.8282528Z 	at io.quarkus.deployment.configuration.BuildTimeConfigurationReader$ReadOperation.readConfigValue(BuildTimeConfigurationReader.java:589)
2020-01-21T11:12:21.8282679Z 	at io.quarkus.deployment.configuration.BuildTimeConfigurationReader$ReadOperation.readConfigGroup(BuildTimeConfigurationReader.java:534)
2020-01-21T11:12:21.8282806Z 	at io.quarkus.deployment.configuration.BuildTimeConfigurationReader$ReadOperation.run(BuildTimeConfigurationReader.java:273)
2020-01-21T11:12:21.8282942Z 	at io.quarkus.deployment.configuration.BuildTimeConfigurationReader.readConfiguration(BuildTimeConfigurationReader.java:230)
2020-01-21T11:12:21.8283073Z 	at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:201)
2020-01-21T11:12:21.8283195Z 	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:85)
2020-01-21T11:12:21.8283314Z 	at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:114)
2020-01-21T11:12:21.8283412Z 	at io.quarkus.dev.DevModeMain.doStart(DevModeMain.java:178)
2020-01-21T11:12:21.8283684Z 	at io.quarkus.dev.DevModeMain.start(DevModeMain.java:96)
2020-01-21T11:12:21.8283775Z 	at io.quarkus.test.QuarkusDevModeTest.beforeEach(QuarkusDevModeTest.java:159)
```

geoand commented:
I was seeing that one as well....

gsmet commented:
Hum. I think it's due to the recent addition of `LocalesBuildTimeConfig` and:
```
    @ConfigItem(defaultValue = "${user.language}-${user.country}", defaultValueDocumentation = "Set containing the build system locale")
    public Set<Locale> locales;
```
I suppose `${user.country}` is not defined on your CI somehow.

lburgazzoli commented:
They are GitHub Actions

gsmet commented:
@lburgazzoli could you try adding `export LANG=en_US` to your GH actions?

gsmet commented:
I'll try to reproduce it locally and come up with a fix in Quarkus.

geoand commented:
@gsmet I can take care of it if you like

gsmet commented:
@geoand ah that would be nice. I think it's just about having `${user.country:}` with a colon.

lburgazzoli commented:
@gsmet you can add them to the git hub actions yamls (.github); or I can do it by changing tha file and push to the PR

gsmet commented:
ah, I will try to push to the PR

geoand commented:
Building `1.2.0.CR1` locally and then I will tryu

gsmet commented:
@geoand the issue is in master too so you can just give master a shot.

I think just having `export LANG=en` should trigger it.

gsmet commented:
BTW, just so you know, we already properly parse `en-` to a valid locale. So it's really about making the country non-mandatory.

geoand commented:
> BTW, just so you know, we already properly parse `en-` to a valid locale. So it's really about making the country non-mandatory.

Yes, exactly

geoand commented:
I can't reproduce this locally

lburgazzoli commented:
```
 LANG=C.UTF-8
LANGUAGE=
LC_CTYPE="C.UTF-8"
LC_NUMERIC="C.UTF-8"
LC_TIME="C.UTF-8"
LC_COLLATE="C.UTF-8"
LC_MONETARY="C.UTF-8"
LC_MESSAGES="C.UTF-8"
LC_PAPER="C.UTF-8"
LC_NAME="C.UTF-8"
LC_ADDRESS="C.UTF-8"
LC_TELEPHONE="C.UTF-8"
LC_MEASUREMENT="C.UTF-8"
LC_IDENTIFICATION="C.UTF-8"
LC_ALL=
```

geoand commented:
I was able to reproduce this locally by `System.clearProperty("user.country")`

lburgazzoli commented:
@geoand @gsmet I'm testing the native images on my laptop to avoid waiting for CI to kicks in, should I merge if everything is OK ?

geoand commented:
> @geoand @gsmet I'm testing the native images on my laptop to avoid waiting for CI to kicks in, should I merge if everything is OK ?

My vote is yes

gsmet commented:
I removed the last commit that was unnecessary. Otherwise, yes, please merge and let's make progress.

lburgazzoli commented:
build in progress

lburgazzoli commented:
going to merge as the build is ok with graalvm 19.2.1  (tested locally)

lburgazzoli commented:
thx @geoand @gsmet

gsmet commented:
Thanks @lburgazzoli .
